src/CSET/__init__.py

```diff
@@ -25,12 +25,45 @@
 from CSET._common import ArgumentError
 
 
-def main():
+def main(raw_cli_args: list[str] = sys.argv):
     """CLI entrypoint.
 
     Handles argument parsing, setting up logging, top level error capturing,
     and execution of the desired subcommand.
     """
+    parser = setup_argument_parser()
```

Comment:
Could you add brief comment on what this is doing so easier for others to understand code when revisiting

Reply:
Added in 29affb6f6b29bc0b1c033463e70cdfa25ed1ae67